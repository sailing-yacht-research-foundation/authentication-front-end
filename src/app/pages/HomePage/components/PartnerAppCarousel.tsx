import React from 'react';
import { Image, Carousel } from 'antd';
import styled from 'styled-components/macro';
import Brand1 from '../assets/brand1.png';
import Brand2 from '../assets/brand2.png';
import Brand3 from '../assets/brand3.png';
import Brand4 from '../assets/brand4.png';
import { useTranslation } from 'react-i18next';
import { translations } from 'locales/translations';

export const PartnerAppCarousel = (props) => {

    const { t } = useTranslation();

    return (
        <Wrapper>
            <h1 className="section-header-text" style={{ textAlign: 'center' }}>{t(translations.home_page.partner_app_carousel.title)}</h1>
            <Carousel slidesToShow={4} dots={{ className: 'carousel-dot partner-app' }} responsive={[
                {
                    breakpoint: 1200,
                    settings: {
                        slidesToShow: 4,
                        slidesToScroll: 1
                    }
                },
                {
                    breakpoint: 1008,
                    settings: {
                        slidesToShow: 3,
                        slidesToScroll: 1
                    }
                },
                {
                    breakpoint: 800,
                    settings: {
                        slidesToShow: 2,
                        slidesToScroll: 1
                    }
                }

            ]} autoplay={true}>
                <ImageContainer style={{ padding: '50px 0' }}>
                    <Image
                        preview={false}
                        width={200}
                        src={Brand1}
                    />
                </ImageContainer>
                
                <ImageContainer>
                    <Image
                        width={200}
                        preview={false}
                        src={Brand2}
                    />
                </ImageContainer>

                <ImageContainer>
                    <Image
                        width={200}
                        preview={false}
                        src={Brand3}
                    />
                </ImageContainer>

                <ImageContainer>
                    <Image
                        width={200}
                        preview={false}
                        src={Brand4}
                    />
                </ImageContainer>

                <ImageContainer>
                    <Image
                        preview={false}
                        width={200}
                        src={Brand1}
                    />
                </ImageContainer>

                <ImageContainer>
                    <Image
                        width={200}
                        preview={false}
                        src={Brand2}
                    />
                </ImageContainer>

                <ImageContainer>
                    <Image
                        width={200}
                        preview={false}
                        src={Brand3}
                    />
                </ImageContainer>

                <ImageContainer>
                    <Image
                        width={200}
                        preview={false}
                        src={Brand4}
                    />
                </ImageContainer>
            </Carousel>
        </Wrapper>
    )
}

const Wrapper = styled.div`
    margin: 88px 0;
    padding: 0 15px;
`;

const ImageContainer = styled.div`
    margin-top: 60px;    
`;