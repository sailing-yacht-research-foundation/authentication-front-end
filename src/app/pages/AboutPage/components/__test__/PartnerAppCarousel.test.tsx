import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { PartnerAppCarousel } from '../PartnerAppCarousel';

const shallowRenderer = createRenderer();

describe('<PartnerAppCarousel />', () => {
  it('should render and matchs the snapshot', () => {
    shallowRenderer.render(<PartnerAppCarousel />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});