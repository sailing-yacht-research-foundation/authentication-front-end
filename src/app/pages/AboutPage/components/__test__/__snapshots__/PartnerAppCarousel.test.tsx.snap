// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PartnerAppCarousel /> should render and matchs the snapshot 1`] = `
<PartnerAppCarousel__Wrapper>
  <h1
    className="section-header-text"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Use with these innotive apps
  </h1>
  <ForwardRef
    autoplay={true}
    dots={
      Object {
        "className": "carousel-dot partner-app",
      }
    }
    responsive={
      Array [
        Object {
          "breakpoint": 1200,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 4,
          },
        },
        Object {
          "breakpoint": 1008,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 3,
          },
        },
        Object {
          "breakpoint": 800,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 2,
          },
        },
      ]
    }
    slidesToShow={4}
  >
    <PartnerAppCarousel__ImageContainer
      style={
        Object {
          "padding": "50px 0",
        }
      }
    >
      <Image
        preview={false}
        src="brand1.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
    <PartnerAppCarousel__ImageContainer>
      <Image
        preview={false}
        src="brand2.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
    <PartnerAppCarousel__ImageContainer>
      <Image
        preview={false}
        src="brand3.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
    <PartnerAppCarousel__ImageContainer>
      <Image
        preview={false}
        src="brand4.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
    <PartnerAppCarousel__ImageContainer>
      <Image
        preview={false}
        src="brand1.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
    <PartnerAppCarousel__ImageContainer>
      <Image
        preview={false}
        src="brand2.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
    <PartnerAppCarousel__ImageContainer>
      <Image
        preview={false}
        src="brand3.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
    <PartnerAppCarousel__ImageContainer>
      <Image
        preview={false}
        src="brand4.png"
        width={200}
      />
    </PartnerAppCarousel__ImageContainer>
  </ForwardRef>
</PartnerAppCarousel__Wrapper>
`;
